LGTM. I've seen a talk about this a while ago, very cool tool and great to see it getting released and added to nixpkgs!

Unfortunately, can't be cross-compiled due to dependencies not supporting cross compilation :(. Nothing to do about it.

<details><summary>Cross build log</summary>

```
rick@nixos-asus:~/tmp/review-poke$ nix build .#pkgsCross.aarch64-multiplatform.poke
error: --- Error --- nix-daemon
builder for '/nix/store/vvaikggwz92z8bmxrg1189z0la0j27p4-expect-5.45.4-aarch64-unknown-linux-gnu.drv' failed with exit code 1; last 10 log lines:
  checking if stty reads stdout... no
  checking if running Solaris... no
  checking for memmove... yes
  checking for sysconf... yes
  checking for strftime... yes
  checking for strchr... yes
  checking for timezone... yes
  checking for siglongjmp... yes
  checking for memcpy... yes
  checking if WNOHANG requires _POSIX_SOURCE... configure: error: Expect can't be cross compiled
error: --- Error --- nix-daemon
1 dependencies of derivation '/nix/store/7rdv71km193rfv2811a802i2kasl1gpn-dejagnu-1.6.2-aarch64-unknown-linux-gnu.drv' failed to build
error: --- Error ----------------------------------------------------------------------------------------------- nix
1 dependencies of derivation '/nix/store/vjbap3pn7lxnrbkc39q0nlzahm52cycv-poke-1.0-aarch64-unknown-linux-gnu.drv' failed to build
```

</details>

--- a/pkgs/applications/editors/poke/default.nix
+++ b/pkgs/applications/editors/poke/default.nix
@@ -1,0 +1,71 @@
+{ lib
+, stdenv
+, fetchurl
+, gettext
+, help2man
+, pkg-config
+, texinfo
+, makeWrapper
+, boehmgc
+, readline
+, guiSupport ? false, tcl, tcllib, tk
+, miSupport ? true, json_c
+, nbdSupport ? true, libnbd
+, textStylingSupport ? true
+, dejagnu
+}:
+
+stdenv.mkDerivation rec {
+  pname = "poke";
+  version = "1.0";
+
+  src = fetchurl {
+    url = "mirror://gnu/${pname}/${pname}-${version}.tar.gz";
+    hash = "sha256-3pMLhwDAdys8LNDQyjX1D9PXe9+CxiUetRa0noyiWwo=";
+  };
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    gettext
+    help2man
+    pkg-config
+    texinfo
+  ] ++ lib.optional guiSupport makeWrapper;
+
+  buildInputs = [ boehmgc dejagnu readline ]
+  ++ lib.optional guiSupport tk
+  ++ lib.optional miSupport json_c
+  ++ lib.optional nbdSupport libnbd
+  ++ lib.optional textStylingSupport gettext;
+
+  configureFlags = lib.optionals guiSupport [
+    "--with-tcl=${tcl}/lib"
+    "--with-tk=${tk}/lib"
+    "--with-tkinclude=${tk.dev}/include"
+  ];
+
+  enableParallelBuilding = true;
+
+  doCheck = true;
+  checkInputs = [ dejagnu ];
+
+  postFixup = lib.optionalString guiSupport ''
+    wrapProgram "$out/bin/poke-gui" \
+      --prefix TCLLIBPATH ' ' ${tcllib}/lib/tcllib${tcllib.version}
+  '';
+
+  meta = with lib; {
+    description = "Interactive, extensible editor for binary data";
+    homepage = "http://www.jemarch.net/poke";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ AndersonTorres metadark ];
+    platforms = platforms.unix;
+  };
+}
+
+# TODO: Enable guiSupport by default once it's more than just a stub

--- a/pkgs/applications/editors/poke/default.nix
+++ b/pkgs/applications/editors/poke/default.nix
@@ -15,7 +15,9 @@
 , dejagnu
 }:
 
-stdenv.mkDerivation rec {
+let
+  isCross = stdenv.hostPlatform != stdenv.buildPlatform;
+in stdenv.mkDerivation rec {
   pname = "poke";
   version = "1.0";
 
@@ -37,11 +39,12 @@ stdenv.mkDerivation rec {
     texinfo
   ] ++ lib.optional guiSupport makeWrapper;
 
-  buildInputs = [ boehmgc dejagnu readline ]
+  buildInputs = [ boehmgc readline ]
   ++ lib.optional guiSupport tk
   ++ lib.optional miSupport json_c
   ++ lib.optional nbdSupport libnbd
-  ++ lib.optional textStylingSupport gettext;
+  ++ lib.optional textStylingSupport gettext
+  ++ lib.optional (!isCross) dejagnu;
 
   configureFlags = lib.optionals guiSupport [
     "--with-tcl=${tcl}/lib"
@@ -51,8 +54,8 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  doCheck = true;
-  checkInputs = [ dejagnu ];
+  doCheck = !isCross;
+  checkInputs = lib.optionals (!isCross) [ dejagnu ];
 
   postFixup = lib.optionalString guiSupport ''
     wrapProgram "$out/bin/poke-gui" \

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -24570,6 +24570,8 @@ in
 
   plugin-torture = callPackage ../applications/audio/plugin-torture { };
 
+  poke = callPackage ../applications/editors/poke { };
+
   polar-bookshelf = callPackage ../applications/misc/polar-bookshelf { };
 
   poezio = python3Packages.poezio;

